### PR TITLE
Bundle FFmpeg for everything

### PR DESCRIPTION
**Pull request by def-**
@heinrich5991 Maybe you want to take a look already?

### CODE REVIEW

**Comment by def-** [comment]
Should I make it shared libs on Linux?

**Comment by def-** [comment]
Mac still fails to build:
```
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
/home/deen/isos/ddnet/ddnet-master/AVCODEC_INCLUDEDIR
   used as include directory in directory /home/deen/isos/ddnet/ddnet-master
/home/deen/isos/ddnet/ddnet-master/AVFORMAT_INCLUDEDIR
   used as include directory in directory /home/deen/isos/ddnet/ddnet-master
/home/deen/isos/ddnet/ddnet-master/AVUTIL_INCLUDEDIR
   used as include directory in directory /home/deen/isos/ddnet/ddnet-master
/home/deen/isos/ddnet/ddnet-master/SWRESAMPLE_INCLUDEDIR
   used as include directory in directory /home/deen/isos/ddnet/ddnet-master
/home/deen/isos/ddnet/ddnet-master/SWSCALE_INCLUDEDIR
   used as include directory in directory /home/deen/isos/ddnet/ddnet-master
AVCODEC_LIBRARY (ADVANCED)
    linked by target "DDNet" in directory /home/deen/isos/ddnet/ddnet-master
AVFORMAT_LIBRARY (ADVANCED)
    linked by target "DDNet" in directory /home/deen/isos/ddnet/ddnet-master
AVUTIL_LIBRARY (ADVANCED)
    linked by target "DDNet" in directory /home/deen/isos/ddnet/ddnet-master
SWRESAMPLE_LIBRARY (ADVANCED)
    linked by target "DDNet" in directory /home/deen/isos/ddnet/ddnet-master
SWSCALE_LIBRARY (ADVANCED)
    linked by target "DDNet" in directory /home/deen/isos/ddnet/ddnet-master
```
Maybe my dylibs are bad? Here's how I built:
```
$ ./configure --disable-all --disable-alsa --disable-iconv --disable-libxcb --disable-libxcb-shape --disable-libxcb-xfixes --disable-sdl2 --disable-xlib --disable-zlib --enable-avcodec --enable-avformat --enable-encoder=libx264,aac --enable-muxer=mp4,mov --enable-protocol=file --enable-libx264 --enable-swresample --enable-swscale --enable-gpl --extra-cflags="-mmacosx-version-min=10.9 -I../x264-master" --extra-cxxflags="-mmacosx-version-min=10.9 -I../x264-master" --extra-ldflags="-L../x264-master" --arch=x86_64 --target_os=darwin --cross-prefix=x86_64-apple-darwin17- --disable-static --enable-shared --cc=$CC --cxx=$CXX                            
install prefix            /usr/local
source path               .
C compiler                o64-clang
C library                 
host C compiler           gcc
host C library            glibc
ARCH                      x86 (generic)
big-endian                no
runtime cpu detection     yes
standalone assembly       yes
x86 assembler             nasm
MMX enabled               yes
MMXEXT enabled            yes
3DNow! enabled            yes
3DNow! extended enabled   yes
SSE enabled               yes
SSSE3 enabled             yes
AESNI enabled             yes
AVX enabled               yes
AVX2 enabled              yes
AVX-512 enabled           yes
XOP enabled               yes
FMA3 enabled              yes
FMA4 enabled              yes
i686 features enabled     yes
CMOV is fast              yes
EBX available             yes
EBP available             yes
debug symbols             yes
strip symbols             yes
optimize for size         no
optimizations             yes
static                    no
shared                    yes
postprocessing support    no
network support           yes
threading support         pthreads
safe bitstream reader     yes
texi2html enabled         no
perl enabled              yes
pod2man enabled           yes
makeinfo enabled          yes
makeinfo supports HTML    yes

External libraries:
appkit                  bzlib                   libx264
avfoundation            coreimage               securetransport

External libraries providing hardware acceleration:
audiotoolbox            cuda_llvm               videotoolbox

Libraries:
avcodec                 avformat                avutil                  swresample              swscale

Programs:

Enabled decoders:

Enabled encoders:
aac                     libx264

Enabled hwaccels:

Enabled parsers:

Enabled demuxers:

Enabled muxers:
mov                     mp4

Enabled protocols:
file

Enabled filters:

Enabled bsfs:
aac_adtstoasc           null                    vp9_superframe

Enabled indevs:

Enabled outdevs:

License: GPL version 2 or later

WARNING: x86_64-apple-darwin17-pkg-config not found, library detection may fail.
WARNING: using libx264 without pkg-config
config.asm is unchanged
libavutil/avconfig.h is unchanged
libavfilter/filter_list.c is unchanged
libavcodec/codec_list.c is unchanged
libavcodec/parser_list.c is unchanged
libavcodec/bsf_list.c is unchanged
libavformat/demuxer_list.c is unchanged
libavformat/muxer_list.c is unchanged
libavdevice/indev_list.c is unchanged
libavdevice/outdev_list.c is unchanged
libavformat/protocol_list.c is unchanged
ffbuild/config.sh is unchanged
$ make -j4
[...]
ld: warning: target OS does not support re-exporting symbol _avpriv_pix_fmt_bps_mov from libavcodec/libavcodec.dylib

ld: warning: target OS does not support re-exporting symbol _avpriv_toupper4 from libavcodec/libavcodec.dylib

ld: warning: target OS does not support re-exporting symbol _avsubtitle_free from libavcodec/libavcodec.dylib

GEN	libswscale/libswscale.ver
LD	libswscale/libswscale.5.dylib
ld: warning: building for macOS, but linking in object file (libswscale/x86/input.o) built for free standing
ld: warning: building for macOS, but linking in object file (libswscale/x86/output.o) built for free standing
ld: warning: building for macOS, but linking in object file (libswscale/x86/rgb_2_rgb.o) built for free standing
ld: warning: building for macOS, but linking in object file (libswscale/x86/scale.o) built for free standing
ld: warning: building for macOS, but linking in object file (libswscale/x86/yuv_2_rgb.o) built for free standing
```

**Comment by def-** [comment]
Ok, got it working now. Will build test versions.

File: cmake/FindFFMPEG.cmake [hunk 1]
```diff
@@ -5,44 +5,56 @@ if(NOT CMAKE_CROSSCOMPILING)
   pkg_check_modules(PC_AVUTIL libavutil)
   pkg_check_modules(PC_SWSCALE libswscale)
   pkg_check_modules(PC_SWRESAMPLE libswresample)
+if(TARGET_OS STREQUAL "linux")
+  pkg_check_modules(PC_X264 libx264)
+endif()
```
Comment (heinrich5991) [inline cmake/FindFFMPEG.cmake:10]: Whitespace.

**Review by heinrich5991 — COMMENTED** [review]
I prefer dynamic linking but I agree that this is a bit late. ;) You seem to prefer static linking. ^^

I guess this is tested already and ready to go?

File: cmake/FindFFMPEG.cmake [hunk 2]
```diff
@@ -125,11 +149,8 @@ is_bundled(IS_BUNDLED "${AVCODEC_LIBRARY}")
 if(IS_BUNDLED AND TARGET_OS STREQUAL "windows")
```
Comment (heinrich5991) [inline cmake/FindFFMPEG.cmake:149]: Rename `IS_BUNDLED` to `FFMPEG_BUNDLED` and the detection should work fine.

**Comment by def-** [comment]
> I guess this is tested already and ready to go?

Yes, tested on Linux, Windows (Wine) and Mac OS X (Learath2). Seems to work exactly as well as with system's ffmpeg libs.

**Comment by heinrich5991** [comment]
bors r+

Nice that we'll get this in the default version. :)

**Pull request merged**